### ISSUE TITLE
GNIS Redirect

### ISSUE THREAD

**Issue by dblodgett-usgs** [internetofwater/geoconnex.us#159]
Now that a GNIS app is available, I think we should go ahead and mint GNIS redirects. I've reached out to the team working on the app and will try and get some dialogue going here.

## Contact Information
Your name: David Blodgett

Your email: dblodgett@usgs.gov

Your organization: USGS

Your organization website: https://edits.nationalmap.gov/apps/gaz-domestic/public/search/names

### The base PID pattern you would like
https://geoconnex.us/usgs/gnis/[0-9]+

PID pattern:

### The desired target URI base pattern
https://edits.nationalmap.gov/apps/gaz-domestic/public/summary/$1

Target pattern:

### Explanation of the kinds of expressions to match from the end of the PID pattern to the end of the target pattern.
Match any number of digits

Expression type:

### 2-3 example target URIs
https://edits.nationalmap.gov/apps/gaz-domestic/public/summary/1577073
https://edits.nationalmap.gov/apps/gaz-domestic/public/summary/204770

target URI example:

Target URI example:

### Link to sitemap or file to create one.
Available for download here: https://www.usgs.gov/u.s.-board-on-geographic-names/download-gnis-data
Available as an ArcGIS map service: https://carto.nationalmap.gov/arcgis/rest/services/geonames/MapServer

**Comment by ksonda** [comment]
Will need to create a template. Since we can proxy the esri service, it *may not be necessary for you to prepare any data, unless you want to make links to other data that can't be generated on the fly from the data in the service.

**Comment by ksonda** [comment]
Yeah on thinking about this more this may be a good opportunity to experiment with generating a json-ld graph for bulk import

**Comment by ksonda** [comment]
starting a repo that will include CI with hydroshare https://github.com/cgs-earth/ref_gnis

**Comment by ksonda** [comment]
when we do this, we should update the GNIS urls in all reference features that were using the Apex urls

**Comment by ksonda** [comment]
@C-Loftus @dblodgett-usgs @webb-ben @adplincinst 

Bumping this issue per a recent email thread with the National Map People.

I think we

1. Mint geoconnex.us/usgs/gnis/<gaz_id> as a regex redirect to https://edits.nationalmap.gov/apps/gaz-domestic/public/summary/<gaz_id>

2. Mint geoconnex.us/iow/gnis/<collection_type>/<gaz_id> to 

- redirect to a pygeoapi we set up with appropriate JSON-LD templates that proxies each of the layers on this esri MapServer https://carto.nationalmap.gov/arcgis/rest/services/geonames/MapServer 
- the JSON-LD templates here include a `owl:sameAs` geoconnex.us/usgs/gnis/<gaz_id>

**Comment by dblodgett-usgs** [comment]
Thumbs up from me!
